dev-notes/minimal-os-for-fobnail.md

```diff
@@ -0,0 +1,203 @@
+porting.
+
+### Xous
+
+Xous is a microkernel-based OS written in Rust, currently in Alpha state. Xous
+is used as a secure microkernel for [Betrusted](betrusted.io) device and is
+focused on RISC-V, which right now is the only architecture supported. However,
+x86 already contains
+[dummy implementation](https://github.com/betrusted-io/xous-core/blob/30b82b25b100e958790973c129dc49e1acca79ec/kernel/src/arch/x86_64.rs)
+which probably will be extended someday.
+
+
+#### Running in DLME
+
+Xous currently cannot run on x86, and DLME on RISC-V is not covered here.
+
+#### Fobnail integration
+
+Xous has network support but currently lacks USB host drivers and TPM support.
+Also, it lacks C interface for calling the kernel and lacks C library. Getting
+[fobnail-attester](https://github.com/fobnail/fobnail-attester) running may
+require a significant amount of work. Also, Xous needs to gain kexec-like
+abilities to chainload target OS.
+
+### seL4
+
+seL4 is a secure L4 family microkernel written in C. It has strong security
+guarantees assured by [Formal proofs](https://sel4.systems/Info/FAQ/proof.pml),
+however these proofs are still incomplete for x86, see
+[Supported Platforms](https://docs.sel4.systems/Hardware) for an up-to-date
+verification status. seL4 due to its microkernel nature provides higher
+isolation, breach in one of the components (like USB driver, network stack)
+wouldn't compromise entire OS contrary to monolithic kernels.
+
+#### Running in DLME
+
+seL4 is a secure L4 family microkernel written in C. It has strong security
+guarantees assured by [Formal proofs](https://sel4.systems/Info/FAQ/proof.pml).
+However, these proofs are still incomplete for x86, see
+[Supported Platforms](https://docs.sel4.systems/Hardware) for an up-to-date
+verification status. seL4, due to its microkernel nature, provides higher
+isolation. A breach in one of the components (like USB driver, network stack)
+wouldn't compromise the entire OS, contrary to monolithic kernels.
+
+#### Fobnail integration
+
+seL4 provides virtually no drivers, except a few drivers listed here.
+According to the page linked above, `libusbdrivers` is inactive and lacks XHCI
+support. seL4 has basic support for an old version of musl C library (v1.1.16).
+
+Using seL4 would require a significant amount of work:
+- Extending USB drivers
+- Implementing USB EEM driver
+- Implementing TPM driver
+- Porting [fobnail-attester](https://github.com/fobnail/fobnail-attester)
+- seL4 runs all processes in unprivileged mode, so the kernel itself would have
+  to be modified to allow booting of another kernel
+
+[Genode](https://github.com/genodelabs/genode) can run on seL4 and provides
+drivers, at least for some devices and a network stack. Its support for seL4
+used to be incomplete, and many components were broken. However, that might have
+changed, and Genode may be an easier way to get software running on seL4.
```

Comment:
@TomaszAIR @arturkow2000 I already reached Sid asking him about review, but maybe asking Genode community would be also good idea. At some point we supported them also, so maybe asking for this small favor would be ok.

Reply:
https://github.com/genodelabs/genode/issues/4480

Reply:
added and mentioned in report